Tools/SandboxTest.ps1

```diff
@@ -83,7 +83,35 @@ New-Item $tempFolder -ItemType Directory -ErrorAction SilentlyContinue | Out-Nul
 $WebClient = New-Object System.Net.WebClient
 
 function Get-Release {
-  $releasesAPIResponse = Invoke-RestMethod 'https://api.github.com/repos/microsoft/winget-cli/releases?per_page=100'
+  # TODO: Further improvements
+  # Sending authorized requests with GitHub CLI (aka. `gh.exe`) is much more convenient than using `Invoke-RestMethod`.
+  # It would be better integrate this script with GitHub CLI, since developers do not need creating and configuring PAT manually.
+
+  # First query an endpoint to see if credentials are valid.
+  $isValidCredential = $false
+  $requestParameters = @{
+    Uri = 'https://api.github.com/octocat'
+  }
+  if ($env:GITHUB_TOKEN) {
+    $requestParameters.Add('Authentication', 'Bearer')
+    $requestParameters.Add('Token', $(ConvertTo-SecureString "$env:GITHUB_TOKEN" -AsPlainText))
+    $isValidCredential = ([Int16](Invoke-WebRequest @requestParameters).Headers["X-RateLimit-Limit"][0]) -eq 60
```

Comment:
This has the potential to throw an exception if that specific header isn’t presnt or can’t be converted to an integer. Either check that the header is present first, or use a different parameter like the HTTP response code (if that changes based on whether or not the user is authenticated)

Reply:
API Endpoint `https://api.github.com/octocat` is able to query without authentication. Users can get an ASCII draw of GitHub Octocat. Both (un)authorized request will get `HTTP 200 OK` from this API.

Maybe switching to [REST API endpoints for rate limits](https://docs.github.com/en/rest/rate-limit/rate-limit?apiVersion=2022-11-28) would be better? Response is in JSON format so I can `ConvertFrom-Json` its `.Content` and get the `[Int64]` type `$.resources.core.limit`.